File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Go/s685734377_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type TaskB struct{}

func (t *TaskB) solve(testNumber int, in *bufio.Reader, out *bufio.Writer) {
	n, _ := strconv.Atoi(readLine(in))
	a := make([][]rune, n)
	for i := 0; i < n; i++ {
		line := readLine(in)
		a[i] = []rune(line)
	}

	P := int64(37)
	Q := int64(47)
	powP := make([]int64, n+1)
	powQ := make([]int64, n+1)
	powP[0] = 1
	powQ[0] = 1
	for i := 1; i <= n; i++ {
		powP[i] = powP[i-1] * P
		powQ[i] = powQ[i-1] * Q
	}

	hashRow := make([][]int64, n)
	hashCol := make([][]int64, n)
	for i := range hashRow {
		hashRow[i] = make([]int64, n)
		hashCol[i] = make([]int64, n)
	}

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			for k := 0; k < n; k++ {
				hashRow[i][k] += int64(a[i][j]) * powP[(j-k+n)%n]
				hashCol[j][k] += int64(a[i][j]) * powP[(i-k+n)%n]
			}
		}
	}

	ans := 0
	for A := 0; A < n; A++ {
		for B := 0; B < n; B++ {
			ok := true
			for i := 0; i < n; i++ {
				if hashRow[(i+A)%n][B] != hashCol[(i+B)%n][A] {
					ok = false
					break
				}
			}
			if ok {
				ans++
			}
		}
	}
	fmt.Fprintln(out, ans)
	out.Flush()
}

func readLine(r *bufio.Reader) string {
	line, _ := r.ReadString('\n')
	return strings.TrimSpace(line)
}

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	solver := &TaskB{}
	solver.solve(1, in, out)
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/C/s685734377_comments_removed.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    FILE *in;
    char *buffer;
    char *next_token;
} FastScanner;

FastScanner* createFastScanner(FILE *in) {
    FastScanner *fs = (FastScanner*)malloc(sizeof(FastScanner));
    fs->in = in;
    fs->buffer = NULL;
    fs->next_token = NULL;
    return fs;
}

int nextInt(FastScanner *fs) {
    return atoi(next(fs));
}

char* next(FastScanner *fs) {
    if (fs->next_token == NULL || *fs->next_token == '\0') {
        if (getline(&fs->buffer, NULL, fs->in) == -1) {
            return NULL;
        }
        fs->next_token = strtok(fs->buffer, " \n");
    } else {
        fs->next_token = strtok(NULL, " \n");
    }
    return fs->next_token;
}

void solve(int n, char a[n][n], long P, long Q, FILE *out) {
    long powP[n + 1], powQ[n + 1];
    powP[0] = powQ[0] = 1;
    for (int i = 1; i <= n; i++) {
        powP[i] = powP[i - 1] * P;
        powQ[i] = powQ[i - 1] * Q;
    }

    long hashRow[n][n], hashCol[n][n];
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            hashRow[i][j] = 0;
            hashCol[i][j] = 0;
        }
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            for (int k = 0; k < n; k++) {
                hashRow[i][k] += a[i][j] * powP[(j - k + n) % n];
                hashCol[j][k] += a[i][j] * powP[(i - k + n) % n];
            }
        }
    }

    int ans = 0;
    for (int A = 0; A < n; A++) {
        for (int B = 0; B < n; B++) {
            int ok = 1;
            for (int i = 0; i < n; i++) {
                if (hashRow[(i + A) % n][B] != hashCol[(i + B) % n][A]) {
                    ok = 0;
                    break;
                }
            }
            ans += ok;
        }
    }
    fprintf(out, "%d\n", ans);
}

int main() {
    FastScanner *in = createFastScanner(stdin);
    FILE *out = stdout;

    int n = nextInt(in);
    char a[n][n];
    for (int i = 0; i < n; i++) {
        char *line = next(in);
        for (int j = 0; j < n; j++) {
            a[i][j] = line[j];
        }
    }

    long P = 37, Q = 47;
    solve(n, a, P, Q, out);

    free(in->buffer);
    free(in);
    fclose(out);
    return 0;
}
// End of Code
